src/logic/otp2/controller.ts

```diff
@@ -204,7 +186,7 @@ function authorityMapper(authority?: Authority): Authority | undefined {
     return {
         id: authority.id,
         name: authority.name,
-        codeSpace: authority.id.split(':')[0],
+        codeSpace: authority.id.split(':')[0] || '',
```

Comment:
@draperunner This is a consequence of turning on strict indexing checks. Not sure '' is the correct here, but I assume it should never happen either.

Reply:
Should the function just return `undefined` if there is no `codeSpace`?

```typescript
function authorityMapper(authority?: Authority): Authority | undefined {
    const codeSpace = authority?.id.split(':')[0]
    if (!authority || ! codeSpace) return undefined
    return {
        id: authority.id,
        name: authority.name,
        codeSpace',
```

Reply:
Yes, I think that's a good idea actually. It shouldn't happen in practice I guess (?)